kinto/integration-test/test_kinto_sync.py

```diff
@@ -0,0 +1,156 @@
+import base64
+import datetime
+import re
+import requests
+import time
+import unittest
+
+from kinto_client import Client
+from kinto_client import replication
+
+class Kinto_Sync(unittest.TestCase):
+    """
+    Tests that given two Kinto servers you can successfully sync data
+    from one to the other
+    """
+
+    def setUp(self):
+        self.master_url = "http://192.168.99.100:8888/v1/"
+        self.read_only_url = "http://192.168.99.100:8889/v1/"
+        self.credentials = ('testuser', 'abc123')
+        self.master = Client(server_url=self.master_url, auth=self.credentials)
+        self.read_only = Client(server_url=self.read_only_url, auth=self.credentials)
+
+        # Try and create our buckets and collections
+        try:
+            self.master.get_bucket('test')
+        except:
+            self.master.create_bucket('test')
+
+        try:
+            self.master.get_collection('test_collection', bucket='test')
+        except:
+            self.master.create_collection('test_collection', bucket='test')
+
+        try:
+            self.read_only.get_bucket('test')
+        except:
+            self.read_only.create_bucket('test')
+
+        try:
+            self.read_only.get_collection('test_collection', bucket='test')
+        except:
+            self.read_only.create_collection('test_collection', bucket='test')
+
+    def test_sync(self):
+        # Generate some random records
+        collection = self.get_timestamp()
+        self.master.create_collection(collection, bucket='test')
+        self.read_only.create_collection(collection, bucket='test')
+        x = 1
+        while x <= 10:
+            self.read_only.create_record(
+                data=self.create_record(),
+                bucket='test',
+                collection=collection)
+            x = x + 1
```

Comment:
more of a judgement call, but pdehaan's suggestion is what i'd go with.
this may help: https://en.wikipedia.org/wiki/Zen_of_Python
dense 1-liners aren't necessarily better, but in this case it actually makes things simpler